framework/src/modules/token/cc_commands/cc_transfer.ts

```diff
@@ -56,99 +53,109 @@ export class CCTransferCommand extends BaseCCCommand {
 
 	public init(args: { minBalances: MinBalance[]; supportedTokenIDs: Buffer[] }): void {
 		this._supportedTokenIDs = args.supportedTokenIDs;
-		this._minBalances = args.minBalances;
+		// this._minBalances = args.minBalances;
 	}
 
-	public async execute(ctx: CCCommandExecuteContext): Promise<void> {
-		const { ccm } = ctx;
+	public async verify(ctx: CCCommandExecuteContext): Promise<void> {
+		const { ccm, ccmSize } = ctx;
 		const methodContext = ctx.getMethodContext();
 		const { id: ownChainID } = await this._interopMethod.getOwnChainAccount(methodContext);
 		let params: CCTransferMessageParams;
-		let tokenChainID;
-		let tokenLocalID;
+		let tokenChainID: Buffer;
+		const { sendingChainID, status, fee } = ccm;
+
 		try {
 			params = codec.decode<CCTransferMessageParams>(crossChainTransferMessageParams, ccm.params);
 			validator.validate(crossChainTransferMessageParams, params);
+			const { tokenID, amount } = params;
+			[tokenChainID] = splitTokenID(tokenID);
+
+			if (
+				!tokenChainID.equals(ownChainID) &&
+				!tokenChainID.equals(sendingChainID) &&
+				!tokenChainID.equals(tokenChainID)
```

Comment:
this should be mainchainID?
mainchainID can be obtained using
```
this._tokenMethod.getMainchainTokenID();
```

Reply:
i think you need to rebase to the feature branch

Reply:
Cool! 😎